packages/patterns/src/patterns/patternMatchers.js

```diff
@@ -1726,50 +1700,138 @@ export const {
-/** @typedef {import('@endo/marshal').CopyTagged} CopyTagged */
-/** @template T @typedef {import('@endo/marshal').CopyRecord<T>} CopyRecord */
-/** @template T @typedef {import('@endo/marshal').CopyArray<T>} CopyArray */
-/** @typedef {import('@endo/marshal').Checker} Checker */
-/** @typedef {import('@endo/marshal').RankCompare} RankCompare */
-/** @typedef {import('@endo/marshal').RankCover} RankCover */
-
-/** @typedef {import('../types').ArgGuard} ArgGuard */
-/** @typedef {import('../types').MethodGuardMaker} MethodGuardMaker */
-/** @typedef {import('../types').MatcherNamespace} MatcherNamespace */
-/** @typedef {import('../types').Key} Key */
-/** @typedef {import('../types').Pattern} Pattern */
-/** @typedef {import('../types').PatternKit} PatternKit */
-/** @typedef {import('../types').CheckPattern} CheckPattern */
-/** @typedef {import('../types').Limits} Limits */
-/** @typedef {import('../types').AllLimits} AllLimits */
-/** @typedef {import('../types').GetRankCover} GetRankCover */
+// //////////////////////////// Guards ///////////////////////////////////////
+
+const AwaitArgGuardShape = harden({
+  klass: 'awaitArg',
+  argGuard: M.pattern(),
+});
+
+export const isAwaitArgGuard = specimen =>
+  matches(specimen, AwaitArgGuardShape);
+harden(isAwaitArgGuard);
+
+export const assertAwaitArgGuard = specimen => {
+  mustMatch(specimen, AwaitArgGuardShape, 'awaitArgGuard');
+};
+harden(assertAwaitArgGuard);
 
 /**
- * @typedef {object} MatchHelper
- * This factors out only the parts specific to each kind of Matcher. It is
- * encapsulated, and its methods can make the stated unchecker assumptions
- * enforced by the common calling logic.
- *
- * @property {(allegedPayload: Passable,
- *             check: Checker
- * ) => boolean} checkIsWellFormed
- * Reports whether `allegedPayload` is valid as the payload of a CopyTagged
- * whose tag corresponds with this MatchHelper's Matchers.
- *
- * @property {(specimen: Passable,
- *             matcherPayload: Passable,
- *             check: Checker,
- * ) => boolean} checkMatches
- * Assuming validity of `matcherPayload` as the payload of a Matcher corresponding
- * with this MatchHelper, reports whether `specimen` is matched by that Matcher.
- *
- * @property {import('../types').GetRankCover} getRankCover
- * Assumes this is the payload of a CopyTagged with the corresponding
- * matchTag. Return a RankCover to bound from below and above,
- * in rank order, all possible Passables that would match this Matcher.
- * The left element must be before or the same rank as any possible
- * matching specimen. The right element must be after or the same
- * rank as any possible matching specimen.
+ * @param {Pattern} argGuard
+ * @returns {AwaitArgGuard}
  */
+const makeAwaitArgGuard = argGuard => {
+  /** @type {AwaitArgGuard} */
+  const result = harden({
+    klass: 'awaitArg',
+    argGuard,
+  });
+  assertAwaitArgGuard(result);
+  return result;
+};
```

Comment:
Would it be worthwhile to try to push the `argPattern` semantics as far down as possible without breaking compatibility?

```suggestion
 * @param {Pattern} argPattern
 * @returns {AwaitArgGuard}
 */
const makeAwaitArgGuard = argPattern => {
  /** @type {AwaitArgGuard} */
  const result = harden({
    klass: 'awaitArg',
    argGuard: argPattern,
  });
  assertAwaitArgGuard(result);
  return result;
};
```

Reply:
Done